File: Java/DesignPatterns/src/main/java/designPatterns/creational/builder/Product.java
package designPatterns.creational.builder;

public class Product {
  private String part1;
  private String part2;

  public void setPart1(String part1) {
    this.part1 = part1;
  }

  public void setPart2(String part2) {
    this.part2 = part2;
  }

  public String show() {
    return "Product with " + part1 + " and " + part2;
  }
}

File: Kotlin/DesignPatterns/src/main/kotlin/creational/builder/Product.kt
package creational.builder

class Product {
  private var _part1: String? = null
  private var _part2: String? = null

  var part1: String?
    get() = _part1
    set(value) {
      _part1 = value
    }

  var part2: String?
    get() = _part2
    set(value) {
      _part2 = value
    }

  fun show(): String {
    return "Product with $_part1 and $_part2"
  }
}
